src/utils.rs

```diff
@@ -97,17 +138,24 @@ pub fn change_device_volume(
 	let (device, device_name): (DeviceInfo, String) = match device_type {
 		VolumeDeviceType::Sink(controller) => {
 			let server_info = controller.get_server_info();
-			let device_name = &match server_info {
-				Ok(info) => info.default_sink_name.unwrap_or("".to_string()),
-				Err(e) => {
-					eprintln!("Pulse Error: {}", e);
-					return None;
-				}
-			};
-			match controller.get_device_by_name(device_name) {
+			let device_name;
+			let global_name = get_device_name();
+			if global_name == "default" {
+				device_name = match server_info {
+					Ok(info) => info.default_sink_name.unwrap_or("".to_string()),
+					Err(e) => {
+						eprintln!("Error getting default_sink: {}", e);
+						return None;
+					}
+				};
+			} else {
+				device_name = global_name;
+				set_device_name("default".to_string());
+			}
```

Comment:
You should be able to simplify this to this:
```rust
let global_name = get_device_name();
let device_name = if global_name == "default" {
	match server_info {
		Ok(info) => info.default_sink_name.unwrap_or("".to_string()),
		Err(e) => {
			eprintln!("Error getting default_sink: {}", e);
			return None;
		}
	};
} else {
	set_device_name("default".to_string());
	global_name;
}
```

Also, don't forget about the SOURCE case ;)

Reply:
All done.

I also noticed an error in the existing version, as you had `default_sink_name` instead of `default_source_name` even if `VolumeDeviceType::Source`. I think no one noticed it as probably few people regularly change their mic volume.